Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm

```diff
@@ -979,6 +981,56 @@ - (void)_webView:(WKWebView *)webview mouseDidMoveOverElement:(_WKHitTestResult
     TestWebKitAPI::Util::run(&done);
 }
 
+static BlockPtr<NSEvent*(NSEvent*)> gEventMonitorHandler;
+
+@interface TestEventMonitor : NSObject
+
++ (id)addLocalMonitorForEventsMatchingMask:(NSEventMask)mask handler:(NSEvent* (^)(NSEvent *event))block;
+
+@end
+
+@implementation TestEventMonitor
+
++ (id)addLocalMonitorForEventsMatchingMask:(NSEventMask)mask handler:(NSEvent* (^)(NSEvent *event))block
+{
+    gEventMonitorHandler = makeBlockPtr(block);
+    return nil;
+}
+
+@end
+
+TEST(WebKit, MouseMoveOverElementWithClosedWebView)
+{
+    auto linkLocation = NSMakePoint(200, 150);
+
+    ClassMethodSwizzler localMonitorSwizzler(NSEvent.class, @selector(addLocalMonitorForEventsMatchingMask:handler:), [TestEventMonitor methodForSelector:@selector(addLocalMonitorForEventsMatchingMask:handler:)]);
+
+    // We swizzle `NSWindow.mouseLocationOutsideOfEventStream` because manually calling the handler intercepted
+    // from the swizzled `+[NSEvent addLocalMonitorForEventsMatchingMask:handler:]` with a fake event means we
+    // skip over the bookkeeping required for this SPI to provide the correct mouse location.
+    InstanceMethodSwizzler mouseLocationSwizzler(NSWindow.class, @selector(mouseLocationOutsideOfEventStream), imp_implementationWithBlock(^{
+        return linkLocation;
+    }));
+
+    @autoreleasepool {
+        WKWebViewConfiguration *configuration = [WKWebViewConfiguration _test_configurationWithTestPlugInClassName:@"FrameHandleSerialization"];
+        auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 400, 300) configuration:configuration]);
```

Comment:
Nit - I think this `WKWebViewConfiguration` (which initializes the "FrameHandleSerialization" plugin) is also unnecessary.
```suggestion
        auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 400, 300)]);
```

Reply:
(Never mind, looks like it's necessary because of `MouseMoveOverElementDelegate`)

Reply:
Yeah, the `MouseMoveOverElementDelegate` in this test file hardcodes a userInfo class check against `_WKFrameHandle`, so we either use this configuration or add a workaround. Staying with this for now then